File: src/org/cappuccino_project/ide/intellij/plugin/psi/utils/ObjJSelectorLiteralPsiUtil.java
package org.cappuccino_project.ide.intellij.plugin.psi.utils;

import org.cappuccino_project.ide.intellij.plugin.psi.ObjJExpr;
import org.cappuccino_project.ide.intellij.plugin.psi.ObjJMethodCall;
import org.cappuccino_project.ide.intellij.plugin.psi.interfaces.ObjJConditionalStatement;

public class ObjJSelectorLiteralPsiUtil {

    public static boolean isRespondToSelector(ObjJMethodCall callTarget) {
        if (true) {
            return false;
        }
        String selectorString = callTarget.getSelectorString();
        for (ObjJConditionalStatement conditionalStatement : ObjJBlockPsiUtil.getParentBlockChildrenOfType(callTarget, ObjJConditionalStatement.class, true)) {
            if (conditionalStatement != null) {
                ObjJExpr expr = conditionalStatement.getConditionalExpression();
                while (expr != null) {
                    if (expr.getLeftExpr() != null) {
                        if (expr.getLeftExpr().getSelectorLiteral() != null && expr.getLeftExpr().getSelectorLiteral().getSelectorString().equals(selectorString)) {
                            return true;
                        }
                    }

                }
            }
        }
        return false;
    }
}

File: src/org/cappuccino_project/ide/intellij/plugin/psi/utils/ObjJSelectorLiteralPsiUtil.kt
package org.cappuccino_project.ide.intellij.plugin.psi.utils

import org.cappuccino_project.ide.intellij.plugin.psi.ObjJExpr
import org.cappuccino_project.ide.intellij.plugin.psi.ObjJMethodCall
import org.cappuccino_project.ide.intellij.plugin.psi.interfaces.ObjJConditionalStatement

object ObjJSelectorLiteralPsiUtil {

    fun isRespondToSelector(callTarget: ObjJMethodCall): Boolean {
        if (true) {
            return false
        }
        val selectorString = callTarget.selectorString
        for (conditionalStatement in ObjJBlockPsiUtil.getParentBlockChildrenOfType(callTarget, ObjJConditionalStatement::class.java, true)) {
            if (conditionalStatement != null) {
                val expr = conditionalStatement.conditionalExpression
                while (expr != null) {
                    if (expr.leftExpr != null) {
                        if (expr.leftExpr!!.selectorLiteral != null && expr.leftExpr!!.selectorLiteral!!.selectorString == selectorString) {
                            return true
                        }
                    }

                }
            }
        }
        return false
    }
}
